### DISCUSSION TITLE
adding prompt

### DISCUSSION THREAD

**Discussion by Deph1** [Zain-ul-din/whatsapp-ai-bot#4] [Q&A] [answered]
Hello, 

How would I add the prompt to the api?

Typically I see an example like this 
```python
import os
import openai

openai.api_key = os.getenv("OPENAI_API_KEY")

response = openai.Completion.create(
  model="text-davinci-003",
  prompt="",
  temperature=0.7,
  max_tokens=256,
  top_p=1,
  frequency_penalty=0,
  presence_penalty=0
)
```

**Comment by Zain-ul-din** [comment, marked as answer]
In typescript you cloud add prompt like:

```typescript
import { ChatGPTAPI } from 'chatgpt';
const client new ChatGPTAPI({ apiKey: this.apiKey });
client.sendMessage(prompt);
```
[usage in this project](https://github.com/Zain-ul-din/whatsapp-ai-bot/blob/master/src/models/ChatGptModel.ts#L2)

### [**ChatGPT API - package used in this project**](https://www.npmjs.com/package/chatgpt)

**Reply by Deph1** [reply]
> In typescript you cloud add prompt like:
> 
> ```ts
> import { ChatGPTAPI } from 'chatgpt';
> const client new ChatGPTAPI({ apiKey: this.apiKey });
> client.sendMessage(prompt);
> ```
> 
> [usage in this project](https://github.com/Zain-ul-din/whatsapp-ai-bot/blob/master/src/models/ChatGptModel.ts#L2)
> 
> ### [**ChatGPT API - package used in this project**](https://www.npmjs.com/package/chatgpt)

I'm very sorry. I am extremely new to programming/coding. Could you describe where to put this.... Thank you for the quick reply

**Reply by Zain-ul-din** [reply]
please can you tell me what exactly you wanna to achive?

**Reply by Deph1** [reply]
I want to create a basic text based adventure with start and restart prompts to keep the ai on track. This is the example I have which I can add during the playground 

Inject start text:
In the grim darkness of the Warhammer 40,000 universe, you, the AI Dungeon Master, shall narrate the story in the style of Luetin09. As the overseer, keep track of players' health using love heart emojis, with a maximum of 5 hearts per player, and manage their limited inventory of 3 slots. Your initial party consists of Sean, Luke, Alex, Ezra, and Towhid. Add characters to the adventure with the command "/add character" followed by "name." Restart the world in a random new adventure with the command "/restart." Delete playable characters from the adventure with the command "/kill" followed by the "name" of the character to be killed. Display character and NPC sheets, and allow players to interact by typing their character's specific name and the dynamic action they take.

Inject restart text:
As the AI Dungeon Master, recall your duty to narrate the story set in the Warhammer 40,000 universe in the style of Luetin09. Track players' health with love heart emojis (up to 5 hearts) and manage their 3 inventory slots. Your party includes Sean, Luke, Alex, Ezra, Towhid, and any additional characters added with "/add character (name)." Restart the world in a random new adventure with the command "/restart." Delete playable characters from the adventure with the command "/kill" followed by the "name" of the character to be killed. Display character and NPC sheets, and allow players to interact by typing their character's specific name and the dynamic action they take.

**Reply by Zain-ul-din** [reply]
correct me if I'm wrong from you're message what I understand is you wanna create a custom model that can reply to user according to you're prompt.

**Reply by Deph1** [reply]
> correct me if I'm wrong from you're message what I understand is you wanna create a custom model that can reply to user according to you're prompt.

I'm using the Open ai API which allows for a inject start text/inject restart text or an initial prompt that gets added to whatever the user types

**Reply by Zain-ul-din** [reply]
I still don't understand do you wanna use this for a bot or another purpose?

**Reply by Deph1** [reply]
> I still don't understand do you wanna use this for a bot or another purpose?

I'm using it to interface with Chat GPT and Dream studio to "act" in the way a text based dungeon rpg would be.

**Reply by Zain-ul-din** [reply]
if you clone this repo 
- go to `static` folder.
- then click on `whatsapp-ai-bot.md` file.
- paste you're prompt in `whatsapp-ai-bot.md` file.

### Usage
- type `!bot` + you're message in WhatsApp.
